PIL/ImageDraw.py

```diff
@@ -366,3 +370,10 @@ def floodfill(image, xy, value, border=None):
                             pixel[s, t] = value
                             newedge.append((s, t))
             edge = newedge
+
+  
+def _color_diff(rgb1, rgb2):
+    """
+    Uses distance formula to calculate difference between two rgb values.
+    """
+    return ((rgb1[0]-rgb2[0])**2 + (rgb1[1]-rgb2[1])**2 + (rgb1[2]-rgb2[2])**2)**.5
```

Comment:
Classic optimization technique, when comparing pythagorean distances many times, compare (dx^2 + dy^2) to thresh^2, instead of taking the square root each time. It's basically a free optimization that doesn't change the outcome. 

I'm not sure of the speed here, but just comparing linear distance is actually a pretty close approximation. (within 1/8 if you take a cube with corners at x,y,z = +-(9/8)*threshold,0 rather than a sphere of radius threshold.  9/8 gives you roughly equal error at the corners and center)